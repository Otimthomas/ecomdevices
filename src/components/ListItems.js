import React from "react";
import styled from "styled-components";

import ListItem from "../components/ListItem";

function ListItems() {
  const items = [
    {
      id: 1,
      image: "/static/iphone.jpg",
      name: "Iphone",
    },
    {
      id: 2,
      image: "/static/laptop.jpg",
      name: "Laptop",
    },
    {
      id: 3,
      image: "/static/printer.jpg",
      name: "Printer",
    },
  ];

  return (
    <Wrapper>
      {items.map((item) => {
        return (
          <div key={item.id}>
            <ListItem {...item} />
          </div>
        );
      })}
    </Wrapper>
  );
}

export default ListItems;

const Wrapper = styled.div`
  padding: 60px 80px;

  display: flex;
  justify-content: center;

  div:not(:first-child) {
    margin-left: 30px;
  }
`;
