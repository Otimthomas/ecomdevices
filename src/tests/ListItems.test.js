import ListItems from "../components/ListItems";
import renderer from "react-test-renderer";

describe("<ListItems />", () => {
  it("matches the snapshot", () => {
    let tree = renderer.create(<ListItems />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
