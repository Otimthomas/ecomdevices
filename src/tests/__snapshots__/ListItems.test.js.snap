// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItems /> matches the snapshot 1`] = `
<div
  className="sc-gsDKAQ dHdcHe"
>
  <div>
    <div
      className="sc-bdvvtL efPRUK"
    >
      <div
        className="card"
      >
        <img
          alt="Iphone"
          src="/static/iphone.jpg"
        />
        <p>
          Iphone
        </p>
      </div>
    </div>
  </div>
  <div>
    <div
      className="sc-bdvvtL efPRUK"
    >
      <div
        className="card"
      >
        <img
          alt="Laptop"
          src="/static/laptop.jpg"
        />
        <p>
          Laptop
        </p>
      </div>
    </div>
  </div>
  <div>
    <div
      className="sc-bdvvtL efPRUK"
    >
      <div
        className="card"
      >
        <img
          alt="Printer"
          src="/static/printer.jpg"
        />
        <p>
          Printer
        </p>
      </div>
    </div>
  </div>
</div>
`;
